android/src/main/java/com/tailscale/ipn/App.kt

```diff
@@ -459,29 +455,15 @@ class App : Application(), libtailscale.AppContext {
 
   @Throws(IOException::class, GeneralSecurityException::class)
   override fun getSyspolicyBooleanValue(key: String): Boolean {
-    val manager = getSystemService(Context.RESTRICTIONS_SERVICE) as RestrictionsManager
-    return BooleanMDMSetting(key, key).getFrom(manager.applicationRestrictions, this)
+    return getSyspolicyStringValue(key) == "true"
   }
 
   @Throws(IOException::class, GeneralSecurityException::class)
   override fun getSyspolicyStringValue(key: String): String {
-    val manager = getSystemService(Context.RESTRICTIONS_SERVICE) as RestrictionsManager
-
-    val alwaysNeverSetting = AlwaysNeverUserDecidesMDMSetting(key, key, false)
-    val showHideSetting = ShowHideMDMSetting(key, key, false)
-    val stringSetting = StringMDMSetting(key, key, false)
-
-    return when {
-      alwaysNeverSetting.keyExists(key) ->
-          alwaysNeverSetting.getFrom(manager.applicationRestrictions, this).value
-      showHideSetting.keyExists(key) ->
-          showHideSetting.getFrom(manager.applicationRestrictions, this).value
-      stringSetting.keyExists(key) ->
-          stringSetting.getFrom(manager.applicationRestrictions, this) ?: ""
-      else -> {
-        Log.d("MDM", "$key is not defined on Android. Returning empty.")
-        ""
-      }
-    }
+    return MDMSettings.allSettingsByKey[key]?.flow?.value?.toString()
```

Comment:
Since we already have the latest values for these settings in stateflows, we can just get them from there (which is also more performant).